src/ComposerRequireChecker/Cli/CheckCommand.php

```diff
@@ -134,6 +135,15 @@ private function getCheckOptions(InputInterface $input): Options
         if (!$fileName) {
             return new Options();
         }
+
+        if (Phar::running() !== '') {
```

Comment:
looking at whole codebase, the `realpath` was applied regardless the execution was under phar or not. do we really need to check phar here ?

Reply:
you're right, the problem wasn't the phar file but the testing with `vfsStream` which does not work when using `realpath` or other low-level file functions.

I rewrote the test to use a temporary file, which should be ok for this special case.